Tests/Services/HealthTests/HealthTests.swift

```diff
@@ -0,0 +1,236 @@
+        try XCTAssertThrowsError(response.message) { error in
+          XCTAssertTrue(error is RPCError)
+          XCTAssertEqual((error as! RPCError).code, .notFound)
+        }
+      }
+    }
+  }
+
+  func testWatchOnKnownService() async throws {
+    try await withHealthClient { (healthClient, healthProvider) in
+      let testServiceDescriptor = ServiceDescriptor(package: "test.package", service: "TestService")
+      let ignoredServiceDescriptor = ServiceDescriptor(
+        package: "package.to.be.ignored",
+        service: "IgnoredService"
+      )
+
+      let statuses: [ServingStatus] = [.serving, .notServing, .serving, .serving, .notServing]
+
+      try healthProvider.updateService(
+        descriptor: testServiceDescriptor,
+        status: statuses[0]
+      )
+
+      try healthProvider.updateService(
+        descriptor: ignoredServiceDescriptor,
+        status: .notServing
+      )
+
+      var message = Grpc_Health_V1_HealthCheckRequest()
+      message.service = testServiceDescriptor.fullyQualifiedService
+
+      try await healthClient.watch(request: ClientRequest.Single(message: message)) { response in
+        var responseStreamIterator = response.messages.makeAsyncIterator()
+
+        for i in 0 ..< statuses.count {
+          let next = try await responseStreamIterator.next()!
+          let expectedStatus = Grpc_Health_V1_HealthCheckResponse.ServingStatus(from: statuses[i])
+
+          XCTAssertEqual(next.status, expectedStatus)
+
+          if i < statuses.count - 1 {
+            try healthProvider.updateService(
+              descriptor: testServiceDescriptor,
+              status: statuses[i + 1]
+            )
+
+            try healthProvider.updateService(
+              descriptor: ignoredServiceDescriptor,
+              status: .notServing
+            )
+          }
+        }
+      }
+    }
+  }
+
+  func testWatchOnUnknownService() async throws {
+    try await withHealthClient { (healthClient, healthProvider) in
+      try healthProvider.updateService(
+        descriptor: ServiceDescriptor(package: "package.to.be.ignored", service: "IgnoredService"),
+        status: .serving
+      )
+
+      let testServiceDescriptor = ServiceDescriptor(package: "test.package", service: "TestService")
+
+      var message = Grpc_Health_V1_HealthCheckRequest()
+      message.service = testServiceDescriptor.fullyQualifiedService
+
+      try await healthClient.watch(request: ClientRequest.Single(message: message)) { response in
+        var responseStreamIterator = response.messages.makeAsyncIterator()
+        var next = try await responseStreamIterator.next()!
+
+        XCTAssertEqual(next.status, .serviceUnknown)
+
+        try healthProvider.updateService(
+          descriptor: testServiceDescriptor,
+          status: .notServing
+        )
+
+        next = try await responseStreamIterator.next()!
+
+        XCTAssertEqual(next.status, .notServing)
+      }
```

Comment:
I don't understand what we're testing here: it's not an "Unknown" service because we updated the status of it to `.notServing`. If anything we should check that the first and only status is `.serviceUnknown`.